### PR TITLE
JS: Support Typescript 5.6

### PR DESCRIPTION
**Pull request by sidshank**
This pull request enables CodeQL analysis of code using TypeScript 5.6 features. Notably, the use of arbitrary import/export specifiers is now supported. Ref [1](https://github.com/tc39/ecma262/pull/2154), [2](https://github.com/github/codeql/pull/17166)

Release announcement here: https://devblogs.microsoft.com/typescript/announcing-typescript-5-6/

### CODE REVIEW

File: javascript/extractor/lib/typescript/package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "name": "typescript-parser-wrapper",
     "private": true,
     "dependencies": {
-        "typescript": "5.5.2"
+        "typescript": "^5.6.2"
```
Comment (erik-krogh) [inline javascript/extractor/lib/typescript/package.json:5]:
```suggestion
        "typescript": "5.6.2"
```

Lets lock it to `5.6.2`.

**Review by erik-krogh — COMMENTED** [review]
Looks good 👍 

The only relevant change seem to be the arbitrary module identifiers. 

Just one minor nit how we specify the version of TypeScript we use.  

I've started some DCA runs. I'll comment here when they are done.

File: javascript/extractor/src/com/semmle/ts/extractor/TypeScriptASTConverter.java [hunk 2]
```diff
@@ -1203,11 +1203,13 @@ private Node convertExportDeclaration(JsonObject node, SourceLocation loc) throw
...
+      JsonElement exportClauseNode = node.get("exportClause");
       boolean hasTypeKeyword = node.get("isTypeOnly").getAsBoolean();
+      boolean isNamespaceExportNode = hasKind(exportClauseNode, "NamespaceExport");
       List<ExportSpecifier> specifiers =
-          hasKind(node.get("exportClause"), "NamespaceExport")
-              ? Collections.singletonList(convertChild(node, "exportClause"))
-              : convertChildren(node.get("exportClause").getAsJsonObject(), "elements");
+          isNamespaceExportNode
+              ? Collections.singletonList(convertNodeAsIdentifier(exportClauseNode.getAsJsonObject()))
+              : convertChildren(exportClauseNode.getAsJsonObject(), "elements");
```
[the remark this answers is not in the document]
Comment (sidshank) [reply javascript/extractor/src/com/semmle/ts/extractor/TypeScriptASTConverter.java:1212]: Will do. I'm taking a look at `convertImportDeclaration` to see if I need to make similar changes there. Once I figure that out (and make the changes if necessary), I'll add tests.

**Comment by sidshank** [comment]
Closing in favour of: https://github.com/github/codeql/pull/17433

**Pull request closed without merging**